src/search.rs

```diff
@@ -0,0 +1,146 @@
+use crate::search_error::SearchError;
+use std::fs;
+use std::path::{Path, PathBuf};
+
+pub fn search(directory: &Path) -> Result<PathBuf, SearchError> {
```

Comment:
I would rename from `search` to `justfile`, and always call it with the full path name, i.e. `search::justfile`.